CMakeLists.txt

```diff
@@ -36,5 +43,10 @@ add_executable(t3 main.cpp
         framework/library/data_structures/list.h)
 
 # Link SDL2 and SDL_image libraries with full paths
-target_link_libraries(t3 PRIVATE ${SDL2_INSTALL_DIR}/lib/libSDL2.so)
-target_link_libraries(t3 PRIVATE ${SDL2_IMAGE_EXTENSION_BUILD_DIR}/install/lib/libSDL2_image.so)
+if (WIN32)
+    target_link_libraries(t3 PRIVATE ${SDL2_INSTALL_DIR}/bin/SDL2.dll -lmingw32 -lSDL2main -lSDL2 -lSDL2_image -mwindows)
+    target_link_libraries(t3 PRIVATE ${SDL2_IMAGE_EXTENSION_BUILD_DIR}/bin/SDL2_image.dll -lmingw32 -lSDL2main -lSDL2 -lSDL2_image -mwindows)
```

Comment:
there is a repetition here. I don't think we need to link libraries twice here. Also, not sure but can you also give it a try without -mwindows . If it works, then we can give it another try without -lmingw32. Maybe things are working now and you don't need them (a big maybe)

Reply:
I've tried without -mwindows and made it work. -lmingw32 flag has to be the first one, or else it throws the @WinMain16 error.